hibernate-core/src/main/java/org/hibernate/hql/spi/id/inline/AbstractInlineIdsDeleteHandlerImpl.java

```diff
@@ -42,6 +43,9 @@ public AbstractInlineIdsDeleteHandlerImpl(
 
 	@Override
 	public String[] getSqlStatements() {
+		if ( deletes.isEmpty() ) {
```

Comment:
Is there possibility that `deletes` is null here? Then a NPE would be thrown. IMHO, lazy initialization might be overused in Hibernate core codebase.

Reply:
solved thanks @NathanQingyangXu